lib/optionalTests/optionalTest_6_2_19.js

```diff
@@ -519,18 +519,47 @@ function calculateEnvironmentalScoreFromMetrics({
     return Number(score.environmentalMetricScore)
   } else {
     const vector = Object.fromEntries(
-      cvss2Mapping.map((entry) => [
-        entry[1],
-        vectorFromVectorString.get(entry[1]) ??
-          /** @type {any} */ (entry[2])[/** @type {any} */ (metrics)[entry[0]]],
-      ])
+      calculateMetricArray({
+        mapping: cvss2Mapping,
+        metrics,
+        vector: vectorFromVectorString,
+      })
     )
     const score = safelyParseCVSSV2Vector(vector)
     if (!score.success) return null
     return score.environmentalMetricScore
   }
 }
 
+/**
+ * This function takes a cvss vector and a metric object and extracts all cvss
+ * values according to the mapping. It does this by first looking up every property
+ * in the `vector`. If the property doesn't exist there but in the metrics objects,
+ * it takes the value from there.
```

Comment:
`there` seems ambiguous
```suggestion
 * in the `vector`. If the property doesn't exist there but in the metrics objects,
 * it takes the value from the corresponding metrics object.